core/arch/arch.c

```diff
@@ -3377,11 +3378,22 @@ dr_mcontext_to_priv_mcontext(priv_mcontext_t *dst, dr_mcontext_t *src)
             dst->pc = src->pc;
         }
         if (TEST(DR_MC_MULTIMEDIA, src->flags)) {
+            /* XXX i#1312: The structure size will double on 64-bit UNIX (possibly
+             * Windows) builds. A corresponding size check be will added with a
+             * future patch.
+             */
             IF_X86_ELSE({ memcpy(&dst->simd, &src->simd, sizeof(dst->simd)); },
                         {
                             /* FIXME i#1551: NYI on ARM */
                             ASSERT_NOT_IMPLEMENTED(false);
                         });
+            if (src->size > sizeof(dr_mcontext_t) - sizeof(dr_opmask_t)) {
```

Comment:
offsetof is the clean way to check a field and avoid having to update the check with each subsequent change.  See e.g. dr_get_os_version().  This type of subtraction also does not handle padding well.

Reply:
See above.